pkg/agent/nodeportlocal/k8s/controller.go

```diff
@@ -112,6 +122,107 @@ func (c *Controller) EnqueueObj(obj interface{}) {
 	c.queue.Add(podKey)
 }
 
+func (c *Controller) checkDeletedSvc(obj interface{}) (*corev1.Service, error) {
+	deletedState, ok := obj.(cache.DeletedFinalStateUnknown)
+	if !ok {
+		return nil, fmt.Errorf("Received unexpected object: %v", obj)
+	}
+	svc, ok := deletedState.Obj.(*corev1.Service)
+	if !ok {
+		return nil, fmt.Errorf("DeletedFinalStateUnknown object is not of type Service: %v", deletedState.Obj)
+	}
+	return svc, nil
+}
+
+func (c *Controller) EnqueueSvcUpdate(oldObj, newObj interface{}) {
+	// In case where the app selector in Service gets updated from one valid selector to another
+	// both set of pods (corresponding to old and new selector) need to be considered
+	newSvc, isSvc := newObj.(*corev1.Service)
+	if !isSvc {
+		var err error
+		newSvc, err = c.checkDeletedSvc(newObj)
+		if err != nil {
+			klog.Errorf("Got error while processing event update: %v", err)
+			return
+		}
+	}
+
+	oldSvc := oldObj.(*corev1.Service)
+	if oldSvc.ResourceVersion == newSvc.ResourceVersion {
+		return
+	}
+
+	podKeys := append(c.getPodsFromService(oldSvc), c.getPodsFromService(newSvc)...)
```

Comment:
Do we need to process Pods in both set or just the removed Pods and the added Pods?

Reply:
considering a case where the service annotation is updated in an existing Service like so
`nodeportlocal.antrea.io/enabled: "true"` -> `nodeportlocal.antrea.io/enabled: "false"`
getPodsFromService() function won't return any pods for the "new" service because of the annotation check, but we would still want to process the pods corresponding to the old service, which had the proper annotation previously.

Reply:
To address this I added a few checks before pushing into the queue, which should also ensure that the two sets are always mutually exclusive, so there should not be a need to check for uniqueness among the two.